### PR TITLE
Revert "Enable noUncheckedIndexedAccess for ODSP driver (#21461)" and disable noUncheckedIndexedAccess

### PR DESCRIPTION
**Pull request by RishhiB**
This reverts commit https://github.com/microsoft/FluidFramework/pull/21664.
[AB#11815](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/11815)

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts [hunk 1]
```diff
@@ -179,9 +179,8 @@ export class OdspSummaryUploadManager {
 		let blobs = 0;
 		const keys = Object.keys(tree.tree);
 		for (const key of keys) {
-			assert(!key.includes("/"), 0x9cd /* id should not include slashes */);
-			// Non null asserting for now, this should be changed to Object.entries
-			const summaryObject = tree.tree[key]!;
+			assert(!key.includes("/"), "id should not include slashes");
```
Comment (sonalideshpandemsft) [inline packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts:182]: why is this assert modified?

File: packages/drivers/odsp-driver/src/compactSnapshotParser.ts [hunk 2]
```diff
@@ -220,7 +214,6 @@ function readTreeSection(node: NodeCore): {
 			trees[path] = result.snapshotTree;
 			if (records.groupId !== undefined) {
 				const groupId = getStringInstance(records.groupId, "groupId should be a string");
-				// Non null asserting since trees[path] is already created
 				trees[path]!.groupId = groupId;
```
Comment (sonalideshpandemsft) [inline packages/drivers/odsp-driver/src/compactSnapshotParser.ts:217]: non null assertion should be removed here as well?

File: packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts (hunk 1 above)
Comment (RishhiB) [reply packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts]: Thank you for the catch, the revert was corrupted, I redid the revert

File: packages/drivers/odsp-driver/src/compactSnapshotParser.ts (hunk 2 above)
Comment (RishhiB) [reply packages/drivers/odsp-driver/src/compactSnapshotParser.ts]: Thank you for the catch, the revert was corrupted, redid the revert

**Review by sonalideshpandemsft — APPROVED** [review]
LGTM :)

**Pull request merged**